### ISSUE TITLE
Exception in DataGrid with runtime type

### ISSUE THREAD

**Issue by ghost** [microsoft/microsoft-ui-xaml#9458]
### Describe the bug

In WINUI we are currently facing an exception when attempting to load the DataGrid component while using runtime-defined types as our data source. The

 exception occurs during the loading process of the component using these types.

When debugging the provided sample project, please ensure that all the options under **'Common Language Runtime Exceptions'** are enabled.
[DataGridException.zip](https://github.com/microsoft/microsoft-ui-xaml/files/14639325/DataGridException.zip)

### Steps to reproduce the bug

1.Run the provided sample.

### Expected behavior

System.ArgumentNullException (Value cannot be null.) will not occur.

### Screenshots

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/113903149/7b7ca8d5-82e0-4a08-9645-a3ac873e821f)

### NuGet package version

None

### Additional context

Any suggestions, or guidance on how to address the exception encountered when using runtime-defined types with the DataGrid would be greatly appreciated.

**Comment by DhivyaBharathi-SF3890** [comment]
Any updates on this?

**Comment by kmgallahan** [comment]
WiniUI doesn't have an in-box DataGrid, so you are using a control that is maintained elsewhere. Probably the Windows Community Toolkit.

**Comment by DhivyaBharathi-SF3890** [comment]
We are using the DataGrid control from the Windows Community Toolkit.

**Comment by kmgallahan** [comment]
@DhivyaBharathi-SF3890 Yes, and that is maintained in the Windows Community Toolkit repo. Issues with that DataGrid should be raised there:

https://github.com/CommunityToolkit/WindowsCommunityToolkit

https://github.com/CommunityToolkit/Windows

**Comment by ManiAkash9** [comment]
@kmgallahan - We raised a new issue in [WindowsCommunityToolKit](https://github.com/CommunityToolkit/WindowsCommunityToolkit)
Issue Link - https://github.com/CommunityToolkit/WindowsCommunityToolkit/issues/4995
